src/contexts/partner/__tests__/PartnerContext.test.tsx

```diff
@@ -0,0 +1,106 @@
+import PartnerRedirectionAlert from '@/app/[locale]/(simulation)/(large-layout-nosticky)/fin/_components/PartnerRedirectionAlert'
+import { getPartnerFromStorage } from '@/helpers/partners/getPartnerFromStorage'
+import { useExportSituation } from '@/hooks/partners/useExportSituation'
+import { useVerifyPartner } from '@/hooks/partners/useVerifyPartner'
+import { useCurrentSimulation } from '@/publicodes-state'
+import '@testing-library/jest-dom'
+import { render, screen, waitFor } from '@testing-library/react'
+import { useRouter, useSearchParams } from 'next/navigation'
+import { PartnerProvider } from '../PartnerContext'
+
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+  useSearchParams: jest.fn(),
+}))
+
+jest.mock('@/helpers/partners/getPartnerFromStorage', () => ({
+  getPartnerFromStorage: jest.fn(),
+}))
+
+jest.mock('@/helpers/partners/removePartnerFromStorage', () => ({
+  removePartnerFromStorage: jest.fn(),
+}))
+
+jest.mock('@/hooks/partners/useExportSituation', () => ({
+  useExportSituation: jest.fn(),
+}))
+
+jest.mock('@/hooks/partners/useVerifyPartner', () => ({
+  useVerifyPartner: jest.fn(),
+}))
+
+jest.mock('@/publicodes-state', () => ({
+  useCurrentSimulation: jest.fn(),
+}))
+
+const setup = ({
+  partner = 'some-partner',
+  progression = 1,
+  situation = { some: 'data' },
+  redirectUrl = '/partner-site',
+  isPartnerVerified = true,
+} = {}) => {
+  const mockPush = jest.fn()
+
+  // Mock hooks
+  ;(useRouter as jest.Mock).mockReturnValue({ push: mockPush })
+  ;(useSearchParams as jest.Mock).mockReturnValue({
+    entries: () =>
+      new Map([
+        ['partner', partner],
+        ['partner-test', 'test'],
+      ]).entries(),
+  })
+  ;(getPartnerFromStorage as jest.Mock).mockReturnValue(null)
```

Comment:
Why do you mock it. sessionStorage is in js-dom

Reply:
retiré !